src/Perms/perm_images.jl

```diff
@@ -92,36 +92,36 @@ else
     function Base.copy(p::Perm)
         imgs = copy(p.images)
         q = typeof(p)(imgs; check = false)
-        if isdefined(p, :inv, :sequentially_consistent)
-            inv_imgs = copy(@atomic(p.inv).images)
+        if isdefined(p, :inv)
+            inv_imgs = copy(p.inv.images)
             q⁻¹ = typeof(p)(inv_imgs; check = false)
-            @atomic q.inv = q⁻¹
-            @atomic q⁻¹.inv = q
+            q.inv = q⁻¹
+            q⁻¹.inv = q
         end
         return q
     end
 
     function Base.inv(σ::Perm)
-        if !isdefined(σ, :inv, :sequentially_consistent)
+        if !isdefined(σ, :inv)
             if isone(σ)
-                @atomic σ.inv = σ
+                σ.inv = σ
             else
                 σ⁻¹ = typeof(σ)(invperm(σ.images); check = false)
                 # we don't want to end up with two copies of inverse σ floating around
```

Comment:
This comment is not valid (doesn't describe the code / not physically possible). There is an atomic annotation for this (`@atomiconce`), but that doesn't actually fix the physics problem of not being able to do 2 writes simultaneously

Reply:
@vtjnash thanks for the comments!

Yeah, now after you pointed this, it seems obvious that I should have done
```julia
@atomic σ⁻¹.inv = σ # fully sets up the "local inverse"
@atomic σ.inv = σ⁻¹ # makes the "local inverse" visible to the global world.
```
so that `σ` is updated (atomically) only once 🤦🏻

I guess reading of `σ.inv` below should be also `@atomic`?

Reply:
It isn't super clear what you want or need here. I think you're just asking for a release store, since the reverse order has the opposite problem since they reference each other. Otherwise you'd have to use a global mutex (to actually implement the comment) and I'm unclear that you actually want that behavior? You also must use atomiconce macro if you want this only happen once, otherwise it will be atomic, but happen potentially as many times as there are threads

Reply:
:release or :monotonic would be fine there, since that is not globally visible memory. You might get a torn read/write if it is a struct that can be inlined which can cause things to get into an inconsistent state. Using `@atomic` explicitly is definitely more correct and better, even if it happens compiles down to the same cpu instructions though. I just note it so you can be alert to whether there are other possible synchronization issues that are being hidden by the seq_cst just making it run slower and being more rare to observe.